### PR TITLE
Fix html text

### PR DESCRIPTION
**Pull request by prashantacharya**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
 import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+
+import Home from './Pages/Home';
+import Story from './Pages/Story';
+import { ROUTES } from './constants';
```
Comment (AsminBudha) [inline src/App.js:6]: Group imports.

File: src/Components/Button/index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+function Button({ children, onClick, ...props }) {
+  return (
+    <button onClick={onClick} {...props}>
+      {children}
+    </button>
+  );
+}
+
+Button.defaultProps = {
```
Comment (AsminBudha) [inline src/Components/Button/index.js:11]: Also add prop types.

File: src/Components/Comment/index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react';
+import Axios from 'axios';
+
+import { BASE_URL } from '../../constants';
+import { AiOutlineUser } from 'react-icons/ai';
```
Comment (AsminBudha) [inline src/Components/Comment/index.js:5]: Group imports.

File: src/Components/Comment/index.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+class Comment extends Component {
+  state = {
+    data: [],
+    isLoading: true,
+  };
+
+  async componentDidMount() {
+    try {
+      const promises = this.props.comments.map((comment) =>
```
Comment (AsminBudha) [inline src/Components/Comment/index.js:17]: Create separate async function. Do not add async in componentDidMount

File: src/Components/Header/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+import './header.css';
+
+function Header() {
+  return (
+    <header>
+      <div className="container">
+        <Link to="/hackernews">
```
Comment (AsminBudha) [inline src/Components/Header/index.js:10]: /hackernews can be accessed from constant file.

File: src/Components/PostList/index.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import { Link } from 'react-router-dom';
+import { GoComment } from 'react-icons/go';
+import { AiOutlineStar, AiOutlineUser } from 'react-icons/ai';
+
+import './postlist.css';
+
+function PostList(props) {
+  return (
+    <li className="story-card">
+      <Link to={`/story/${props.post.id}`}>
```
Comment (AsminBudha) [inline src/Components/PostList/index.js:11]: .

File: src/Context/StoriesContext.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    isLoading: false,
+  };
+
+  fetchDataFromId = async () => {
+    const promises = [];
+
+    const startIndex = this.state.selectedPage * POSTS_PER_PAGE;
+    const endIndex = startIndex + POSTS_PER_PAGE;
+
+    for (let i = startIndex; i < endIndex; i++) {
```
Comment (AsminBudha) [inline src/Context/StoriesContext.js:23]: Can also be used declarative array functions.

File: src/Pages/Home/index.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import { BsChevronCompactLeft, BsChevronCompactRight } from 'react-icons/bs';
+
+import Button from '../../Components/Button';
+import { POSTS_PER_PAGE } from '../../constants';
+import PostList from '../../Components/PostList';
```
Comment (AsminBudha) [inline src/Pages/Home/index.js:6]: .

File: src/Components/PostList/index.js (hunk 6 above)
Comment (prashantacharya) [reply src/Components/PostList/index.js:11]: How can that be done? I have /story/:id on constants but I need to pass id here.

File: src/Components/Comment/index.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+  fetchComments = async () => {
+    try {
+      const promises = this.props.comments.map((comment) =>
+        Axios.get(`${BASE_URL}/item/${comment}.json`)
+      );
+
+      const response = await Promise.all(promises);
+      const data = await response.map((res) => res.data);
+      this.setState({ data, isLoading: false });
```
Comment (AsminBudha) [inline src/Components/Comment/index.js:24]: Leave line before.

File: src/Components/Comment/index.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    } catch (error) {
+      console.log(error);
```
Comment (AsminBudha) [inline src/Components/Comment/index.js:26]: Remove console.log. Or you can use console.error

File: src/Components/Comment/index.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  };
+
+  componentDidMount() {
+    this.fetchComments();
+  }
+
+  render() {
+    return (
+      <div className="comments-wrapper">
+        {this.state.isLoading && <p style={{ textAlign: 'center' }}>Loading</p>}
```
Comment (AsminBudha) [inline src/Components/Comment/index.js:37]: Separate inline css.

File: src/Components/Footer/index.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+function Footer() {
+  return (
+    <footer>
+      <div className="container">
+        Created By &nbsp;
+        <a
+          href="https://github.com/prashantacharya"
+          target="_blank"
+          rel="noopener noreferrer"
```
Comment (AsminBudha) [inline src/Components/Footer/index.js:13]: title attribute.

File: src/Components/Footer/index.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
```
Comment (AsminBudha) [inline src/Components/Footer/index.js:1]: Folder name should be camelCase i.e fooBar.

Only component fileName can be uppercase i.e FooBar.js

Also, create Footer.js and export from index.js.

File: src/Components/Header/index.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+import './header.css';
+import { ROUTES } from '../../constants';
```
Comment (AsminBudha) [inline src/Components/Header/index.js:5]: Leave extra line above.

File: src/Components/PostList/index.js (hunk 6 above)
Comment (AsminBudha) [reply src/Components/PostList/index.js:11]: stroryPath.replace(':id',yourId).

File: src/Components/PostList/index.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+        <h2>{props.post.title}</h2>
+      </Link>
+
+      <ul className="story-details">
+        <li>
+          <AiOutlineStar /> {props.post.score}
```
Comment (AsminBudha) [inline src/Components/PostList/index.js:17]: Use object destructuring.

File: src/Context/StoriesContext.js [hunk 16]
```diff
@@ -0,0 +1,87 @@
+import React, { createContext, Component } from 'react';
+import Axios from 'axios';
```
Comment (AsminBudha) [inline src/Context/StoriesContext.js:2]: Leave extra line before.

File: src/Pages/Home/home.css [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+.controls {
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 10px;
+}
+
+.control-text {
+  font-size: 22px;
```
Comment (AsminBudha) [inline src/Pages/Home/home.css:12]: Create variables for font-size.

File: src/Pages/Story/index.js [hunk 18]
```diff
@@ -0,0 +1,68 @@
+import React, { useContext, Fragment } from 'react';
+import { GoComment } from 'react-icons/go';
```
Comment (AsminBudha) [inline src/Pages/Story/index.js:2]: Extra line above.

File: src/Pages/Story/index.js [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+import Comment from '../../Components/Comment';
+import { StoriesContext } from '../../Context/StoriesContext';
+
+import './story.css';
+
+function Story() {
+  const match = useRouteMatch();
+  const context = useContext(StoriesContext);
+  const data = context.data.data;
+  const post = data.filter((element) => element.id === +match.params.id)[0];
```
Comment (AsminBudha) [inline src/Pages/Story/index.js:15]: +match.params.

Why +?

**Pull request merged**